### PR TITLE
Make invoice number padding configurable

### PR DESCRIPTION
**Pull request by devadattas**
Closes https://github.com/FOSSBilling/FOSSBilling/issues/2275

### CODE REVIEW

File: src/modules/Invoice/Service.php [hunk 1]
```diff
@@ -164,14 +164,17 @@ public function toApiArray(\Model_Invoice $invoice, $deep = true, $identity = nu
         }
         $tax = $tax_total;
 
+        $invoice_number_zero_padding = $this->di['mod_service']('system')->getParamValue('invoice_number_zero_padding');
+        $invoice_number_zero_padding = !empty($invoice_number_zero_padding) ? $invoice_number_zero_padding : 5;
```
Comment (BelleNottelling) [inline src/modules/Invoice/Service.php:168]: It's worth mentioning that in the unlikely case someone wants to have zero padding, the usage of `empty` would prevent that and instead result in 5 being used.

Test code:
```PHP
if(empty('0')){
	echo "0 is empty";
} else {
	echo "0 is not empty";
}
```

Gives:
> 0 is empty

Although not as pretty & readable, this would ensure people can specify zero padding if they desired:
```PHP
$invoiceNumberPadding = $invoiceNumberPadding !== null && $invoiceNumberPadding !== '' ? $invoiceNumberPadding : 5;
```

File: src/modules/Invoice/html_admin/mod_invoice_settings.html.twig [hunk 2]
```diff
@@ -85,6 +85,12 @@
                         <input class="form-control" type="text" name="invoice_starting_number" value="{{ params.invoice_starting_number }}">
                     </div>
                 </div>
+                <div class="mb-3 row">
+                    <label class="col-md-3 col-form-label">{{ 'Invoice number zero padding width'|trans }}</label>
```
Comment (BelleNottelling) [inline src/modules/Invoice/html_admin/mod_invoice_settings.html.twig:89]: I think "Invoice number zero padding width" is a bit wordy and might not be the most clear to some users.

I'd probably change it to say "Invoice number padding length" and add a short description under it giving a bit more detail on what it actually does

File: src/modules/Invoice/html_admin/mod_invoice_settings.html.twig [hunk 3]
```diff
@@ -85,6 +85,12 @@
...
+                    <div class="col-md-6">
+                        <input class="form-control" type="number" name="invoice_number_zero_padding" value="{{ params.invoice_number_zero_padding|default(5) }}" min="1" max="20">
```
Comment (BelleNottelling) [inline src/modules/Invoice/html_admin/mod_invoice_settings.html.twig:91]: Let's shorten the variable name down to `invoice_number_padding`

**Review by BelleNottelling — CHANGES_REQUESTED** [review]
A few quite minor comments, but overall this looks like a nice addition for the handful who may need it

File: src/modules/Invoice/Service.php (hunk 1 above)
Comment (devadattas) [reply src/modules/Invoice/Service.php]: Resolved https://github.com/devadattas/FOSSBilling/commit/d38f037320d1e224c449db3fcdc3f059f3571b5a

File: src/modules/Invoice/html_admin/mod_invoice_settings.html.twig (hunk 2 above)
Comment (devadattas) [reply src/modules/Invoice/html_admin/mod_invoice_settings.html.twig]: Resolved with https://github.com/devadattas/FOSSBilling/commit/0bf8c921c9b174431ca305459a3776532af01102

File: src/modules/Invoice/html_admin/mod_invoice_settings.html.twig (hunk 3 above)
Comment (devadattas) [reply src/modules/Invoice/html_admin/mod_invoice_settings.html.twig]: Resolved with https://github.com/devadattas/FOSSBilling/commit/0bf8c921c9b174431ca305459a3776532af01102

**Review by BelleNottelling — APPROVED** [review]
LGTM, thanks!
